sdk/core/core/src/az_span.c

```diff
@@ -643,39 +643,77 @@ AZ_NODISCARD az_result _az_span_url_encode(az_span destination, az_span source,
   int32_t const source_size = az_span_size(source);
   _az_PRECONDITION_VALID_SPAN(destination, source_size, false);
 
-  int32_t result_size = 0;
-  for (int32_t i = 0; i < source_size; ++i)
+  if (source_size == 0)
   {
-    result_size += _az_span_url_should_encode(az_span_ptr(source)[i]) ? 3 : 1;
+    *out_length = 0;
+    return AZ_OK;
   }
 
-  if (az_span_size(destination) < result_size)
+  int32_t const destination_size = az_span_size(destination);
+  if (destination_size < source_size)
```

Comment:
Similarly here, I think we can remove this check (especially given it is already checked as a precondition), since we won't overflow anything even if we let this pass through.

Reply:
I'd keep it with the same reasoning as above.